paddle/fluid/operators/mkldnn/interpolate_v2_mkldnn_op.cc

```diff
@@ -0,0 +1,152 @@
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+#include "paddle/fluid/framework/data_layout_transform.h"
+#include "paddle/fluid/operators/interpolate_v2_op.h"
+#include "paddle/fluid/platform/mkldnn_reuse.h"
+
+namespace paddle {
+namespace operators {
+
+using framework::DataLayout;
+using dnnl::memory;
+using dnnl::primitive;
+using dnnl::reorder;
+using dnnl::stream;
+using dnnl::resampling_forward;
+using platform::GetMKLDNNFormat;
+using platform::to_void_cast;
+
+template <typename T = float>
+class InterpolateV2MKLDNNKernel : public framework::OpKernel<T> {
+  std::vector<int> ComputeOutputShape(
+      const framework::ExecutionContext& ctx) const {
+    const auto* x = ctx.Input<Tensor>("X");
+    auto in_dims = x->dims();
+    const bool is_channel_last = false;  // In mkldnn kernel, always use NCHW
+
+    framework::DDim in_dhw_dims;
+    if (is_channel_last) {  // NDHWC, NHWC, NWC
+      in_dhw_dims = framework::slice_ddim(in_dims, 1, in_dims.size() - 1);
+    } else {  // NCDHW, NCHW, NCW
+      in_dhw_dims = framework::slice_ddim(in_dims, 2, in_dims.size());
+    }
+
+    std::vector<int> out_dims;
+    if (in_dhw_dims.size() == 1) {
+      out_dims.push_back(ctx.Attr<int>("out_w"));
+    } else if (in_dhw_dims.size() == 2) {
+      out_dims.push_back(ctx.Attr<int>("out_h"));
+      out_dims.push_back(ctx.Attr<int>("out_w"));
+    } else if (in_dhw_dims.size() == 3) {
+      out_dims.push_back(ctx.Attr<int>("out_d"));
+      out_dims.push_back(ctx.Attr<int>("out_h"));
+      out_dims.push_back(ctx.Attr<int>("out_w"));
+    }
+
+    auto list_new_size_tensor = ctx.MultiInput<framework::Tensor>("SizeTensor");
+    auto out_size = ctx.Input<Tensor>("OutSize");
+    if (list_new_size_tensor.size() > 0) {
+      auto new_size = get_new_shape(list_new_size_tensor);
+      if (new_size.size() == out_dims.size()) {
+        out_dims = new_size;
+      }
+    } else if (out_size != nullptr) {
+      auto out_size_data = get_new_data_from_tensor<int>(out_size);
+      if (out_size_data.size() == out_dims.size()) {
+        out_dims = out_size_data;
+      }
+    } else {
+      std::vector<float> scales;
+      auto scale_tensor = ctx.Input<Tensor>("Scale");
+      if (scale_tensor != nullptr) {
+        auto scale_data = get_new_data_from_tensor<float>(scale_tensor);
+        std::copy(scale_data.begin(), scale_data.end(), scales.begin());
+        while (scale_data.size() < 3) scales.push_back(scale_data[0]);
+      } else {
```

Comment:
Hi, Is this else branch only part that is different from interpolate_mkldnn_v1?
Maybe function `std::is_same` for type comparison could be used for this case?